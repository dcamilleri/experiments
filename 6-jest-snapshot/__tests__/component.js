import React from 'react'
import renderer from 'react-test-renderer'
import CustomForm from '../component'

const renderJsonComponent = (props) => {
  return renderer.create(
    <CustomForm {...props} />
  ).toJSON()
}

const onSubmit = () => {
  console.log('Hey')
}

const inputs = [
  {
    type: 'text',
    name: 'foo',
    className: 'foo',
    label: 'foo',
    placeholder: 'foo'
  }
]

describe('CustomForm', () => {
  it('should render correctly without any props', () => {
    const comp = renderJsonComponent()
    expect(comp).toMatchSnapshot()
  })

  it('should render correctly with onSubmit', () => {
    const comp = renderJsonComponent({
      onSubmit
    })
    expect(comp).toMatchSnapshot()
  })

  it('should render correctly with inputs', () => {
    const comp = renderJsonComponent({
      inputs
    })
    expect(comp).toMatchSnapshot()
  })

  it('should render correctly all props', () => {
    const comp = renderJsonComponent({
      inputs,
      onSubmit
    })
    expect(comp).toMatchSnapshot()
  })
})
