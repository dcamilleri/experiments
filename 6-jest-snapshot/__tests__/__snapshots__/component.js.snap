// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomForm should render correctly all props 1`] = `
<div
  className="CustomForm"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="Input-ctn"
    >
      <label
        htmlFor="foo"
      >
        foo
      </label>
      <input
        className="Input foo"
        name="foo"
        placeholder="foo"
        type="text"
      />
    </div>
  </form>
</div>
`;

exports[`CustomForm should render correctly with inputs 1`] = `
<div
  className="CustomForm"
>
  <form
    onSubmit={undefined}
  >
    <div
      className="Input-ctn"
    >
      <label
        htmlFor="foo"
      >
        foo
      </label>
      <input
        className="Input foo"
        name="foo"
        placeholder="foo"
        type="text"
      />
    </div>
  </form>
</div>
`;

exports[`CustomForm should render correctly with onSubmit 1`] = `
<div
  className="CustomForm"
>
  <form
    onSubmit={[Function]}
  />
</div>
`;

exports[`CustomForm should render correctly without any props 1`] = `
<div
  className="CustomForm"
>
  <form
    onSubmit={undefined}
  />
</div>
`;
